src/views/EditDetails.js

```diff
@@ -0,0 +1,85 @@
+import axios from 'axios'
+import React from 'react'
+import config from '../config'
+import FormFieldRow from '../components/FormFieldRow'
+import ErrorHandler from '../components/ErrorHandler'
+
+const emailValidRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+
+class EditDetails extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      isRequestFailed: false,
+      requestFailedMessage: '',
+      newAffiliation: '',
+      newEmail: '',
+      newName: '',
+    }
+
+    this.handleOnChange = this.handleOnChange.bind(this)
+    this.isAllValid = this.isAllValid.bind(this)
+    this.handleOnSubmit = this.handleOnSubmit.bind(this)
+  }
+
+  handleOnChange (field, value) {
+    // parent class change handler is always called with field name and value
+    this.setState({ [field]: value })
+  }
+
+  isAllValid () {
+    return true
+  }
+
+  handleOnSubmit (event) {
+    if (!this.isAllValid()) {
+      event.preventDefault()
+      return
+    }
+
+    const request = {
+      newEmail: this.state.newEmail,
+      newAffiliation: this.state.newAffiliation,
+      newName: this.state.newName
+    }
+
+    axios.post(config.api.getUriPrefix() + '/user/editDetails', request)
+      .then(res => {
+        window.location.href = '/'
```

Comment:
`EditDetails.js` is within the switch of `react-router` usage, (we realize now,) so you can change this `window.location.href = '/'` to `this.props.history.push('/')` instead, please.

Reply:
Makes sense. Done!